updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -91,47 +91,70 @@ def dependencies
 
         sig { params(dependency: Dependabot::Dependency).void }
         def check_and_create_pr_with_error_handling(dependency)
-          check_and_create_pull_request(dependency)
-        rescue URI::InvalidURIError => e
-          error_handler.handle_dependency_error(error: Dependabot::DependencyFileNotResolvable.new(e.message),
-                                                dependency: dependency)
-        rescue Dependabot::InconsistentRegistryResponse => e
-          error_handler.log_dependency_error(
-            dependency: dependency,
-            error: e,
-            error_type: "inconsistent_registry_response",
-            error_detail: e.message
-          )
-        rescue StandardError => e
-          process_dependency_error(e, dependency)
+          # Set has_update to nil to indicate that we don't know if there is an update yet
+          has_update = T.let(nil, T.nilable(T::Boolean))
+
+          begin
+            # has_update indicates the result of checking for updates:
+            # - true if an update is available
+            # - false if no update is needed
+            # - nil if an error occurred
+            has_update = check_and_create_pull_request(dependency)
+          rescue URI::InvalidURIError => e
+            error_handler.handle_dependency_error(
+              error: Dependabot::DependencyFileNotResolvable.new(e.message),
+              dependency: dependency
+            )
+          rescue Dependabot::InconsistentRegistryResponse => e
+            error_handler.log_dependency_error(
+              dependency: dependency,
+              error: e,
+              error_type: "inconsistent_registry_response",
+              error_detail: e.message
+            )
+          rescue StandardError => e
+            process_dependency_error(e, dependency)
+          ensure
+            # Record ecosystem metrics if:
+            # - an update is available (has_update = true), or
+            # - an error occurred (has_update = nil)
+            service.record_ecosystem_meta(dependency_snapshot.ecosystem) if has_update != false
+          end
         end
```

Comment:
**Review Tip:** Send metrics for `update all` after update process (success, or fail). Just being sure that metrics is getting called when there is update. We don't prefer to send metrics when there is not update necessary